manifests/repo/debian.pp

```diff
@@ -62,4 +62,35 @@
       }
     }
   }
+
+  if ($php::globals::php_version == '7.1') {
+    # Required packages for PHP 7.1 repository
+    package { 'apt-transport-https':
```

Comment:
Many modules solve this by having a "manage_repo" parameter in the init class.

You should probably also use ensure_packages to avoid conflicts with existing profiles.